keras/src/export/litert.py

```diff
@@ -0,0 +1,548 @@
+import logging
+import os
+import traceback
+
+from keras.src import tree
+from keras.src.utils import io_utils
+from keras.src.utils.module_utils import litert
+from keras.src.utils.module_utils import tensorflow as tf
+
+
+def export_litert(
+    model,
+    filepath,
+    verbose=True,
+    input_signature=None,
+    aot_compile_targets=None,
+    **kwargs,
+):
+    """Export the model as a LiteRT artifact for inference.
+
+    Args:
+        model: The Keras model to export.
+        filepath: The path to save the exported artifact.
+        verbose: `bool`. Whether to print a message during export. Defaults to
+            `None`, which uses the default value set by different backends and
+            formats.
+        input_signature: Optional input signature specification. If
+            `None`, it will be inferred.
+        aot_compile_targets: Optional list of LiteRT targets for AOT
+        compilation.
+        **kwargs: Additional keyword arguments passed to the exporter.
+    """
+
+    exporter = LiteRTExporter(
+        model=model,
+        input_signature=input_signature,
+        verbose=verbose,
+        aot_compile_targets=aot_compile_targets,
+        **kwargs,
+    )
+    exporter.export(filepath)
+    if verbose:
+        io_utils.print_msg(f"Saved artifact at '{filepath}'.")
+
+
+class LiteRTExporter:
+    """Exporter for the LiteRT (TFLite) format.
+
+    This class handles the conversion of Keras models for LiteRT runtime and
+    generates a `.tflite` model file. For efficient inference on mobile and
+    embedded devices, it creates a single callable signature based on the
+    model's `call()` method and supports optional Ahead-of-Time (AOT)
+    compilation for specific hardware targets.
+    """
+
+    def __init__(
+        self,
+        model,
+        input_signature=None,
+        verbose=False,
+        aot_compile_targets=None,
+        **kwargs,
+    ):
+        """Initialize the LiteRT exporter.
+
+        Args:
+            model: The Keras model to export
+            input_signature: Input signature specification
+            verbose: Whether to print progress messages during export.
+            aot_compile_targets: List of LiteRT targets for AOT compilation
+            **kwargs: Additional export parameters
+        """
+        self.model = model
+        self.input_signature = input_signature
+        self.verbose = verbose
+        self.aot_compile_targets = aot_compile_targets
+        self.kwargs = kwargs
```

Comment:
Just for clarity, let's call it `self.tf_lite_converter_kwargs`.

Also, it's very unfortunate that it's not validated right here. If it fails later, you're no longer in `__init__` and the connection between `__init__` and the invalid kwarg is unclear.